### PR TITLE
Create Locations table

### PR DESCRIPTION
**Pull request by pedrosino**
Part of #732

### CODE REVIEW

**Comment by pedrosino** [comment]
I initially thought of having a separate checkbox for allowing notifications for each location.
@jonatanklosko thinks if someone is inserting a location, they would want to be notified, so a general checkbox would make more sense.

Thoughts?

**Comment by pedrosino** [comment]
I think Jonatan's idea makes more sense.
Here's what I thought:

User can select notifications at Preferences tab
image

And set locations at the new Locations tab
image
(just shows current locations now)

File: WcaOnRails/db/migrate/20180206211650_add_locations_table.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class AddLocationsTable < ActiveRecord::Migration[5.1]
+  def up
+    create_table :locations do |t|
+      t.integer :user_id, null: false
+      t.integer :latitude
+      t.integer :longitude
+      t.integer :notification_radius
```
Comment (jfly) [inline WcaOnRails/db/migrate/20180206211650_add_locations_table.rb]: I think this should have a unit associated with it. Is this miles? km?

File: WcaOnRails/db/migrate/20180206211650_add_locations_table.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      t.timestamps
+    end
+
+    add_column :users, :competition_notifications_enabled, :boolean
+  end
+
+  def down
+    drop_table :locations
```
Comment (jfly) [inline WcaOnRails/db/migrate/20180206211650_add_locations_table.rb]: Do you really have to define the down separately here? This feels like the kind of simple migration that rails would be able to revers if you defined it in a `def change` instead.

File: WcaOnRails/app/models/location.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  def latitude_degrees
+    latitude ? latitude / 1e6 : nil
+  end
+
+  def latitude_radians
+    to_radians latitude_degrees
+  end
+
+  # private def compute_coordinates
```
Comment (jfly) [inline WcaOnRails/app/models/location.rb]: Vestigial code?

File: WcaOnRails/db/migrate/20180206211650_add_locations_table.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class AddLocationsTable < ActiveRecord::Migration[5.1]
+  def up
+    create_table :locations do |t|
+      t.integer :user_id, null: false
+      t.integer :latitude
+      t.integer :longitude
```
Comment (jfly) [inline WcaOnRails/db/migrate/20180206211650_add_locations_table.rb]: In retrospect, I regret that we named these latitude and longitude. I think we're actually storing these in microdegrees, so how about `latitude_microdegrees` and `longitude_microdegrees` instead?

**Comment by pedrosino** [comment]
Fixed @jfly comments and changed this PR to contain only the migration and table stuff.
UI and others will be done in another PR.

**Pull request merged**